codecs/rubixDataEncoding/encoder.go

```diff
@@ -183,31 +183,77 @@ func EncodeRequestMessage(points []*model.Point) ([]byte, error) {
 			return nil, err
 		}
 
-		addressID, err := utils.SafeDereferenceUint8(point.AddressID)
+		// IoNumber consists of 2 parts, the first part is point type (i.e. UO, DO, UI, DI, UVP, DVP) and the
+		// second part is the 1-based point number (e.g. 1, 2, 3). These are separated by a "-", e.g. "UO-1", "UVP-5"
+		// We need to extract those 2 parts to construct the position flag for the point.
+		parts := strings.Split(point.IoNumber, "-")
+		if len(parts) != 2 {
+			return nil, errors.New("invalid IoNumber format")
+		}
+		pointType := parts[0]
+		pointNumber := parts[1]
+		position, err := generateRdePosition(pointType, pointNumber)
```

Comment:
maybe lets move this into a func similar/opposite to [generateFieldName()](https://github.com/NubeIO/module-core-loraraw/blob/b9cc5e5029d3b9230002984e882661fc9f31c36e/codecs/rubixDataEncoding/decoder.go#L205C6-L205C23) so that they are documented next to each other since they are related and opposites